DESIGN.md

```diff
@@ -7,90 +7,64 @@
-    - `Call<List<List<Span>>> getTraces(QueryRequest request);`
-    - `Call<List<Span>> getTrace(String traceId);`
-    - `Call<List<String>> getServiceNames();`
-    - `Call<List<String>> getSpanNames(String serviceName);`
-    - `Call<List<DependencyLink>> getDependencies(long endTs, long lookback);`
-* `SpanConsumer`: which ingest spans
-    - `Call<Void> accept(List<Span> spans)`
-* `QueryRequest`: which includes
-    - `String serviceName, spanName;`
-    - `Map<String, String> annotationQuery;`
-    - `Long minDuration, maxDuration;`
-    - `long endTs, lookback;`
-    - `int limit;`
+* More focused on supporting processing than storage: traces and dependency links are emitted 
+downstream to support metrics aggregation. Storage is currently supported but in a single node.
 
 ### Kafka Zipkin Storage
 
-#### `KafkaSpanStore`
-
-Span Store is expecting Spans to be stored in topics partitioned by `TraceId`.
-
-> These can be created by Span Consumer, or can be **enriched** by other Stream Processors, outside of
-Zipkin Server.
-
-Kafka Span Store will need to support different kind of queries:
-
-
-##### Get Service Names/Get Span Names
-
-Service name to Span names pairs are indexed by aggregating spans.
-
-##### Get Trace/Find Traces
-
-When search requests are received, span index is used to search for trace ids. After a list is 
-retrieved, trace DAG is retrieved from trace state store.
+#### `KafkaSpanConsumer`
 
-##### Get Dependencies
+This component take batches of `spans` collected on different transports (e.g. HTTP, Kafka) .
 
-After `spans` are aggregated into traces, traces are processed to collect dependencies. 
-Dependencies changelog are stored in a Kafka topic to be be stored as materialized view on 
-Zipkin instances.
+In order to support aggregation and post-processing, batches are `flatmap`ing into individual 
+spans, keyed by `trace ID`.
 
-### Stream processors
+#### Stream processors
 
 #### Trace Aggregation Stream Processor
 
-This is the main processors that take incoming spans and aggregate them into:
-
+This processor take incoming partitioned spans and aggregate them into:
 - Traces
 - Dependencies
 
-![service aggregation](docs/service-aggregation-stream.png)
+**Trace completion:** A key part of the stream processing pipeline is define when a trace is completed
+to emit its complete state. This is done by using [Session windows](https://kafka.apache.org/23/javadoc/org/apache/kafka/streams/kstream/SessionWindows.html)
+grouping spans by trace ID, and waiting for a certain inactivity period without receiving messages 
+to emit a trace.
 
-![dependency aggregation](docs/dependency-aggregation-stream.png)
+> One **main consideration** with this approach is that in order to evaluate inactivity gap, a new 
+> event has to be received. This means if you send a set of trace spans and wait for it to be shown 
+> as query result, it requires another span passed the inactivity gap to be trigger.
```

Comment:
"This means if you send a set of trace spans and wait for it to be shown  as query result, it requires another span passed the inactivity gap to be trigger."

means... "The session representing the trace can only close due to inactivity upon a span event. The span doesn't need to be in the same trace to close sessions unrelated to it". ?

Reply:
rephrased with links to source code and default names :)